packages/neon-core/src/rpc/RPCClient.ts

```diff
@@ -108,11 +150,13 @@ export class RPCClient {
 
   /**
    * Gets the block at a given height or hash.
+   * @param param: `indexOrHash`-height or hash of desired block; `verbose`- 0 for serialized block in hex, 1 for json format, defaults to 0
+   * @returns string or block object according to verbose
    */
-  public async getBlock(
-    indexOrHash: string | number,
-    verbose = 1
-  ): Promise<object | string> {
+  public async getBlock<T extends GetBlockParam>(
+    param: T
+  ): Promise<getBlockResult<T>> {

```

Comment:
You have mushed up the args in order to get the typing right. Instead of doing this, you should overload the function definitions [instead](https://stackoverflow.com/questions/43958889/return-different-type-by-string-in-typescript).

```
function e(indexOrHash: string, verbose:0): string;
function e(indexOrHash: string): string;
function e(indexOrHash: string, verbose:1): o;
function e(indexOrHash: string, verbose?:0|1): string|o {
    if(verbose === 0) {
        return "hi";
    } else {
        return {h: "hi"}
    }
}
```

The alternative is to ask C# to change their json rpc to return something more consistent. Im of the view that a standard shape request/response structure is a lot easier to work with across languages. We can do this in javascript but it can become a mess in other languages.